### PR TITLE
Fleshes out RankD tests

### LINKED ISSUE #435: Improve RankD Tests
Right now the Rank1D and Rank2D tests are very basic and can be improved using the new assert image similarity and pytest testing framework mechanisms. 

### Proposal/Issue

The test matrix should against the following:

 - [x] replace `make_regression` dataset with `load_energy` and update those tests
 - [x] use `load_occupancy` for classification tests
 - [x] algorithms: pearson, covariance, spearman (2D) and shaprio (1D)
 - [x] 1D case - horizontal orientation
 - [x] 1D case - vertical orientation
 - [x] Test that an exception is raised for unrecognized algorithms
 - [x] test the underlying rank matrix is correct

Unfortunately, we can't use `pytest.mark.parametrize` with visual test cases (yet), so we'll have to make individual tests for each. 

### Code Snippet

Tests will look approximately like:

```python
def test_rank2d_bad_algorithm(self):
    """"
    Assert that unknown algorithms raise exception
    """"
    with pytest.raises(YellowbrickValueError, match="unknown algorithm"):
         # do the thing 

def test_rank2d_pearson_regression(self):
    """"
    Test Rank2D images similar with pearson scores on regression dataset 
    """"
    data = load_energy(return_dataset=True)
    oz = Rank2D(algorithm='pearson')
    oz.fit_transform(data) 
    npt.assert_array_equal(oz.ranks_, [[]])
    self.assert_images_similar(oz, tol=0.25) 
```

### Background

See #68 and #429

### PR DESCRIPTION
**Pull request by rebeccabilbro**
This PR fixes #435, fleshing out test cases for our `Rank1D` and `Rank2D` visualizers.

I have made the following changes:

 - [x] replaced `make_regression` dataset (removing `Dataset` fixture) with `load_energy` and updated all of the affected tests (_note: this necessitated updating several baseline images_)
 - [x] added `Rank1D` horizontal orientation case
 - [x] added a test that an exception is raised for unrecognized algorithms for both `Rank1D` and `Rank2D` 

### CHECKLIST

- [x] _Is the commit message formatted correctly?_
- [x] _Do all of your functions and methods have docstrings?_
- [x] _Have you added/updated unit tests where appropriate?_
- [x] _Have you updated the baseline images if necessary?_
- [x] _Have you run the unit tests using `pytest`?_
- [x] _Is your code style correct (are you using PEP8, pyflakes)?_

### CODE REVIEW

File: tests/test_features/test_rankd.py [hunk 1]
```diff
@@ -74,97 +49,29 @@ def test_kendalltau(self):
...
-                    1.0,
-                ],
+                [1.0, -1.0, -0.2724275, -0.7361443, 0.7385489, 0.0, 0.0, 0.0],
+                [-1.0, 1.0, 0.2724275, 0.7361443, -0.7385489, 0.0, 0.0, 0.0],
+                [-0.2724275, 0.2724275, 1.0, -0.15192004, 0.19528337, 0.0, 0.0, 0.0],
+                [-0.73614431, 0.73614431, -0.15192004, 1.0, -0.87518995, 0.0, 0.0, 0.0],
+                [0.73854895, -0.73854895, 0.19528337, -0.87518995, 1.0, 0.0, 0.0, 0.0],
+                [0.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0],
+                [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.15430335],
+                [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.15430335, 1.0],
```
Comment (bbengfort) [inline tests/test_features/test_rankd.py:63]: Much more compact and nicer looking!

File: tests/test_features/test_rankd.py [hunk 2]
```diff
@@ -306,16 +235,24 @@ def test_rank1d_quick_method(self):
...
     Test the Rank2D visualizer
     """
+    def test_rank2d_unknown_algorithm(self):
+        """
+        Test that an error is raised for Rank2D with an unknown algorithm
+        """
+        X, _ = load_energy()
+        msg = "'oscar' is unrecognized ranking method"
+        with pytest.raises(YellowbrickValueError, match=msg):
+            Rank2D(algorithm="Oscar").transform(X)
```
Comment (bbengfort) [inline tests/test_features/test_rankd.py:255]: easter egg!

File: tests/test_features/test_rankd.py [hunk 3]
```diff
@@ -430,98 +368,99 @@ def test_rank2d_covariance(self):
...
+                    2.40547588e00,
                 ],
             ]
         )
 
         assert hasattr(oz, "ranks_")
-        assert oz.ranks_.shape == (self.dataset.shape[1], self.dataset.shape[1])
-        npt.assert_array_almost_equal(oz.ranks_, expected)
+        assert oz.ranks_.shape == (X.shape[1], X.shape[1])
+        npt.assert_array_almost_equal(oz.ranks_, expected, decimal=5)
```
Comment (bbengfort) [inline tests/test_features/test_rankd.py:463]: Is the decimal parameter related to the numba issue we were having on miniconda?

Comment (rebeccabilbro) [reply tests/test_features/test_rankd.py:463]: I'm not sure; I was noticing local failures with pytest with the default parameter value for `decimal` (which is 6), which were resolved by considering only 5 decimal places.

Comment (bbengfort) [reply tests/test_features/test_rankd.py:463]: Ah, ok - if it was local then it's not related. 5 decimal points are perfectly reasonable for array almost equal in this case; I just wanted to collate and collect related issues.

**Pull request merged**